src/modm/driver/display/st7586s_protocol.hpp

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (c) 2021, Tomasz Wasilczyk
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#pragma once
+#ifndef MODM_ST7586S_HPP
+#error "Do not include this file directly. Include st7586s.hpp instead."
+#endif
+
+#include <modm/math/utils/endianness.hpp>
+
+namespace modm::impl::st7586s
```

Comment:
Typically the naming convention is `modm::detail::st7586s` similarly as in the std lib.

Reply:
Done.